raco/backends/sql/catalog.py

```diff
@@ -107,10 +107,12 @@ def _convert_zeroary_expr(self, cols, expr, input_scheme):
 
     def _convert_unary_expr(self, cols, expr, input_scheme):
         input = self._convert_expr(cols, expr.input, input_scheme)
-        if isinstance(expr, expression.MAX):
-            return func.max(input)
-        if isinstance(expr, expression.MIN):
-            return func.min(input)
+
+        fname = expr.__class__.__name__.lower()
```

Comment:
Yeah, I don't think we can leverage the dialect support here, unfortunately. How about this: we start with your original approach of testing that the function name is directly supported in SQLAlchemy, and if not, then we special-case by dialect (for now, the only supported dialect is PostgreSQL)? I'm sure you can come up with a nice abstraction for delegating the dialect-driven transformation of the function.

The other question is how to pass the dialect to `SQLCatalog.get_sql()`. I can think of two approaches: we could introduce a new keyword arg `dialect` in `SQLCatalog.__init__()`, or we could add a keyword arg `dialect` to  `SQLCatalog.get_sql()` (defaulting to PostgreSQL in both cases, I guess). What do you think?